pkg/planner/core/find_best_task.go

```diff
@@ -1261,6 +1261,10 @@ func (ds *DataSource) FindBestTask(prop *property.PhysicalProperty, planCounter
 		planCounter.Dec(1)
 		return nil, 1, nil
 	}
+	if prop.TaskTp == property.MppTaskType && ds.preferStoreType&h.PreferTiKV > 0 {
```

Comment:
Can't add `prop.TaskTp == Cop && ds.preferStoreType&TiFlash` here, which might lead to `can't find plan` error since TiFlash can't ensure it can always generate a valid plan.